gobblin-core-base/src/main/java/org/apache/gobblin/source/extractor/extract/FlushingExtractor.java

```diff
@@ -108,8 +108,8 @@
   protected Long flushIntervalMillis;
 
   protected Long timeOfLastFlush = System.currentTimeMillis();
-  private FlushAckable lastFlushAckable;
-  private boolean hasOutstandingFlush = false;
+  protected FlushAckable lastFlushAckable;
```

Comment:
See the comment below about creating a shutdown() method in FlushingExtractor class.